### ISSUE TITLE
Add environment to Cloudflare adapter

### ISSUE THREAD

**Issue by piperswe** [sveltejs/kit#2844]
### Describe the problem

Cloudflare Workers provide an environment object containing environment variables and bindings to other Cloudflare services, such as Workers KV and Durable Objects. Currently, the Cloudflare adapter doesn't seem pass this object to endpoints, so those endpoints can't access those services.

### Describe the proposed solution

The Cloudflare adapter should pass the environment object through to the endpoint somehow, potentially through the request's locals. In addition, it would be great to have a way to export Durable Object classes.

### Importance

would make my life easier

**Comment by piperswe** [comment]
Related to https://github.com/sveltejs/kit/issues/2807

**Comment by tv42** [comment]
Not being able to access Cloudflare KV makes implementing sessions way too hard :-(

**Comment by samuel-bach** [comment]
@tv42 yeah it is a pity. For now I'm using KV via [API](https://api.cloudflare.com/#workers-kv-namespace-read-key-value-pair) which works but is of course cumbersome

**Comment by samuba** [comment]
@tv42 yeah it is a pity. For now I'm using KV via [API ](https://api.cloudflare.com/#workers-kv-namespace-read-key-value-pair) in my project. It works but is cumbersome.

**Comment by denizkenan** [comment]
I had to fork existing adapter and pass env to request object. It is a pity that, this adapter even exists. Without access to ENV, Durable Objects using workers is almost pointless.

**Comment by eric-naguras** [comment]
We need this, I hope it will be fixed soon.

**Comment by bryanwood** [comment]
> I had to fork existing adapter and pass env to request object. It is a pity that this adapter even exists(it is pre-alpha IMHO). Without providing access to ENV & Durable Objects, using workers is almost pointless.

This no longer works since the commit that added a check for the keys being passed in from the adapter:

https://github.com/sveltejs/kit/blob/f89fb524f851b794cceba4a91864d127dc3948ee/packages/kit/src/core/build/build_server.js#L91-L94

As a hacky workaround I'm adding `env` to the headers object.

In the forked adapter:
```
const rendered = await app.render({
        url,
        rawBody: new Uint8Array(await req.arrayBuffer()),
        headers: Object.fromEntries([...req.headers, ["env", env]]),
        method: req.method,
      });
```

In hooks.js

```
export const handle = async ({ request, resolve }) => {
	request.locals.env = request.headers['env'];
	delete request.headers['env'];
	// etc ...
}
```

I imagine passing extra keys on the request object would begin working again once that TODO to remove the check is resolved. 

Not sure if passing extra keys to the request object is going to be a reliable way of passing platform features along to hooks.js and endpoints, guessing it's not specifically part of the app.render api contract?

**Comment by benmccann** [comment]
This should be implemented now

https://github.com/sveltejs/kit/tree/master/packages/adapter-cloudflare#environment-variables
